### ISSUE TITLE
HiddenBar not showing all icons on main (notched) display, but shows all on external monitor

### ISSUE THREAD

**Issue by francisconoriega** [dwarvesf/hidden#228]
### Describe the bug

Quite a few of my icons are missing on the main display, even though I'm using the setting to use all the bar. They do show on my external monitor:

### Steps to reproduce
Main Laptop Monitor (MBP16", M1) 
<img width="1720" alt="image" src="https://user-images.githubusercontent.com/4259909/189701368-a51a7a12-03ba-4772-8ffb-cccc3b3df054.png">
### Expected behaviour

External Monitor:

![image](https://user-images.githubusercontent.com/4259909/189701599-13548ce8-df1a-4388-aad7-eacef85c6d2f.png)

(closeup of the right corner)
![image](https://user-images.githubusercontent.com/4259909/189701671-1f5e043c-03e5-49ba-a71c-0e84143301b9.png)

### Environment

<img width="205" alt="image" src="https://user-images.githubusercontent.com/4259909/189701775-faa262ee-4a8b-4e47-8182-cf8aa67fed70.png">

Using Raycast to add Spotify menu icon

**Comment by GrayJack** [comment]
I'm having the same issue

**Comment by kytta** [comment]
I don't think it's a Hidden Bar bug. I guess macOS simply hides icons that don't fit because of the notch 🤷‍♂️

**Comment by FutureSharks** [comment]
If `hidden` opened the "hidden" icons below the menu bar, like Bartender does, it would solve this for users who had a notch on their display:

![image](https://user-images.githubusercontent.com/5198883/192111509-25ae163f-a7d5-4955-96fc-9d113be5bd99.png)

**Comment by flumiie** [comment]
> If `hidden` opened the "hidden" icons below the menu bar, like Bartender does, it would solve this for users who had a notch on their display:
> 
> ![image](https://user-images.githubusercontent.com/5198883/192111509-25ae163f-a7d5-4955-96fc-9d113be5bd99.png)

This is a good idea. I've been using Pro M1 for couple years and recently switched to Air M2. I do also use [RDM](https://github.com/avibrazil/RDM) to extend my resolution beyond the built-in Settings, so I can get away with this for now.

**Comment by iamsoum** [comment]
Yes please @viettrungphan please add this feature from Bartender, if you can. Currently the hidden items go beneath the notch and are invisible.

**Comment by viettrungphan** [comment]
This can't implement with sandbox env, which is the reason why Bartender can't be public in the app store.

**Comment by gabrielwhite** [comment]
I have a suggestion on how to solve this that might be easier / quicker from a SW development perspective (I would contribute myself if I had the skill).

What if clicking on the "<" instead of expanding the icons list to the left did two things:

1. Hide all the "permanent icons" (thus giving more real estate for the hidden ones)
2. Showed the hidden ones in place of the permanent icons.

This would be only a small behaviour change from the current app and likely solve the problems with the notch for most users.

**Comment by earthsound** [comment]
> This can't implement with sandbox env, which is the reason why Bartender can't be public in the app store.

@viettrungphan iBar has this functionality and it's [in the App Store](https://apps.apple.com/us/app/ibar-menubar-icon-control-tool/id6443843900). It's using [Accessibility and Screen Recording](https://www.better365.cn/ibar/authority.html) to do this.

**Comment by earthsound** [comment]
Duplicate of #225
